intelmq/lib/mixins/http.py

```diff
@@ -0,0 +1,106 @@
+""" HttpMixin for IntelMQ
+
+SPDX-FileCopyrightText: 2021 Birger Schacht
+SPDX-License-Identifier: AGPL-3.0-or-later
+
+Based on `create_request_session` in intelmq.lib.utils and
+`set_request_parameters` in intelmq.lib.bot.Bot
+"""
+
+from intelmq.lib.exceptions import MissingDependencyError
+
+try:
+    import requests
+except ImportError:
+    requests = None
+
+
+class TimeoutHTTPAdapter(requests.adapters.HTTPAdapter):
+    """
+    A requests-HTTP Adapter which can set the timeout generally.
+    """
+
+    def __init__(self, *args, timeout=None, **kwargs):
+        self.timeout = timeout
+        super().__init__(*args, **kwargs)
+
+    def send(self, *args, **kwargs):
+        kwargs['timeout'] = self.timeout
+        return super().send(*args, **kwargs)
+
+
+class HttpMixin:
+    """
+    Setup a request session
+    """
+    __session: requests.Session = None
+    __auth = None
+    __proxy = {}
+
+    ssl_client_cert = None  # TODO: path?
+    http_header: dict = {}
+    http_user_agent: str = "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36"
+    http_verify_cert: bool = True
+    http_proxy = None
+    https_proxy = None
+    http_timeout_max_tries: int = 3
+    http_timeout_sec: int = 30
+    http_username = None
+    http_password = None
+
+    def __init__(self, **kwargs):
+        self.logger.debug("Running HTTP Mixin initialization.")
+        self.http_header = {}
+
+    def setup(self):
```

Comment:
That almost duplicates the code in `intelmq.lib.utils.create_request_session`

Reply:
Thats the goal ;)